### DISCUSSION TITLE
Can I use Bud as a replacement htmx?

### DISCUSSION THREAD

**Discussion by Vad1mo** [livebud/bud#114] [Q&A] [answered]
I am looking for something like the [tetraframework](https://www.tetraframework.com/) but in golang, so I was wondering if I could use bud in a similar way as tetra does with alpine?

**Comment by matthewmueller** [comment, marked as answer]
Hey @Vad1mo, thanks for your interest in Bud!

Did you miss a word in the title? "Can I use Bud as a"? 😅 

My answer for this is similar to what I mentioned in the [discussion on Hotwire](https://github.com/livebud/bud/discussions/113#discussioncomment-2874095). Please have a read and let me know what you think.

**Reply by Vad1mo** [reply]
The thing is, that the [tetraframework](https://www.tetraframework.com/) does not need Hotwire/htmx.

**Reply by Vad1mo** [reply]
This example summarizes the advantages, and his is where I wonder if Bud could be used:

From: https://www.tetraframework.com/docs/introduction#todolist-component

> We also create a "public attribute" named title; the value of this is available to both your server side code / template and to your front end JavaScript / Alpine.js.
>There is also a load method - this is called both when initially rendering the component, and when the component is "resumed" from its saved state. (More on this later)

**Reply by matthewmueller** [reply]
How do you add client-side logic with Tetra? Say I wanted to add a character limit that counted down as I'm entering the todo (similar to Twitter).

**Reply by Vad1mo** [reply]
in the same way as HTM, you have a ` js` and `css`, variable in your component. It follows the vue.js component like structure.

```javascript
    script: javascript = """
    export default {
        lastTitleValue: "",
        inputDeleteDown() {
            this.lastTitleValue = this.title;
        },
        inputDeleteUp() {
            if (this.title === "" && this.lastTitleValue === "") {
                // If the input was all ready empty when we pressed backspace then
                // delete the to do item.
                this.delete_item()
            }
        }
    }
    """
```

**Reply by matthewmueller** [reply]
I had a closer look at Alpine.js tonight. Pretty neat concept! I'd definitely need to play with it some more to get a better sense of how well it scales (conceptually). What I like about Svelte / React / etc. is the abstraction is a pretty solid one that most FE developers are familiar with at this point. Definitely comes at a cost of needing a JS evaluator on the server though.

Since Alpine is just adding more logic into HTML, Bud will actually be able to support this use case once we support HTML and `html/template` files as views. You could just add Alpine.js as a script to a layout file and you're ready to go.
